src/Microsoft.OData.Core/JsonLight/JsonLightODataAnnotationWriter.cs

```diff
@@ -124,5 +220,16 @@ private string SimplifyODataAnnotationName(string annotationName)
         {
             return enableWritingODataAnnotationWithoutPrefix ? annotationName.Substring(ODataAnnotationPrefixLength) : annotationName;
         }
+
+        /// <summary>
+        /// Asserts that the annotation writer was created for asynchronous operation.
+        /// </summary>
+        [Conditional("DEBUG")]
+        private void AssertAsynchronous()
+        {
+#if DEBUG
+            Debug.Assert(this.asyncJsonWriter != null, "The method should only be called on an asynchronous annotation writer.");
+#endif

```

Comment:
at least check the "asyncJsonWriter != null"?

Reply:
That is what the assertion statement is doing. I'm not sure I understand your question here @xuzhg

Reply:
It's only in debug mode, I mean why do you have #if DEBUG, then have Debug.Assert(...)?

Debug.Assert() only runs for #if DEBUG, right?

So, remove #if DEBUG?

Reply:
@xuzhg I finally understood your comment.
We're applying the `Conditional("DEBUG")` attribute to this method and it will not be compiled unless the `DEBUG` compilation symbol is defined.
For that reason, I believe we don't need the `#if DEBUG/#endif` statements.
Just to be sure I won't be shooting myself in the foot by removing them, would you know why we've had this practice of applying the `#if DEBUG/#endif` statements in methods with similar `ConditionalAttribute`?
- https://github.com/OData/odata.net/blob/5fc3f2d24c50c6108c053763279a92377eebc099/src/Microsoft.OData.Core/ODataInputContext.cs#L582
- https://github.com/OData/odata.net/blob/5fc3f2d24c50c6108c053763279a92377eebc099/src/Microsoft.OData.Core/ODataInputContext.cs#L594
- https://github.com/OData/odata.net/blob/a355bf73d87bcfff6d1d62d9847b3d304464cb43/src/Microsoft.OData.Core/ODataOutputContext.cs#L662
- https://github.com/OData/odata.net/blob/a355bf73d87bcfff6d1d62d9847b3d304464cb43/src/Microsoft.OData.Core/ODataOutputContext.cs#L674

Reply:
I don't have such history.